lib/tests/streamlit/streamlit_test.py

```diff
@@ -61,48 +58,96 @@ def test_get_option(self):
-    def test_run_warning_presence(self):
-        """Using Streamlit without `streamlit run` produces a warning."""
-        with self.assertLogs(level=logging.WARNING) as logs:
-            st._use_warning_has_been_displayed = False
-            st.write("Using delta generator")
-            output = "".join(logs.output)
-            # Warning produced exactly once
-            self.assertEqual(len(re.findall(r"streamlit run", output)), 1)
-
-    @patch("streamlit.runtime.Runtime.exists", MagicMock(return_value=True))
-    def test_run_warning_absence(self):
-        """Using Streamlit through the CLI results in a Runtime being instantiated,
-        so it produces no usage warning."""
-        with self.assertLogs(level=logging.WARNING) as logs:
-            st._use_warning_has_been_displayed = False
-            st.write("Using delta generator")
-            # assertLogs is being used as a context manager, but it also checks
-            # that some log output was captured, so we have to let it capture something
-            get_logger("root").warning("irrelevant warning so assertLogs passes")
-            self.assertNotRegex("".join(logs.output), r"streamlit run")
+    @parameterized.expand(
+        [
+            # DeltaGenerator methods:
+            "altair_chart",
+            "area_chart",
+            "audio",
+            "balloons",
+            "bar_chart",
+            "bokeh_chart",
+            "button",
+            "caption",
+            "camera_input",
+            "checkbox",
+            "code",
+            "columns",
+            "tabs",
+            "container",
+            "dataframe",
+            "date_input",
+            "download_button",
+            "expander",
+            "pydeck_chart",
+            "empty",
+            "error",
+            "exception",
+            "file_uploader",
+            "form",
+            "form_submit_button",
+            "graphviz_chart",
+            "header",
+            "help",
+            "image",
+            "info",
+            "json",
+            "latex",
+            "line_chart",
+            "map",
+            "markdown",
+            "metric",
+            "multiselect",
+            "number_input",
+            "plotly_chart",
+            "progress",
+            "pyplot",
+            "radio",
+            "selectbox",
+            "select_slider",
+            "slider",
+            "snow",
+            "subheader",
+            "success",
+            "table",
+            "text",
+            "text_area",
+            "text_input",
+            "time_input",
+            "title",
+            "vega_lite_chart",
+            "video",
+            "warning",
+            "write",
+            "color_picker",
+            # Other modules the user should have access to:
+            "echo",
+            "spinner",
+            "set_page_config",
+            "stop",
+            "cache",
```

Comment:
`session_state`, `secrets`, `sidebar` is missing here.

Reply:
I assume the reason is that `session_state`, `secrets`, `sidebar` are not functions/callables. So, we probably need to test their existence differently.

Reply:
Correct - these are not callables (and neither is experimental_user)